src/web/js/ide.js

```diff
@@ -31,32 +48,19 @@ const runtimeApi = {
     // TODO(joe): pass in a URL to uniquely identify this program
     if(!url) { url = "definitions://"; }
     var runtime = window.CPOIDEHooks.runtime;
-    function getVal(mod, name) {
-      return runtime.getField(runtime.getField(mod, "values"), name);
-    }
-    var parse = getVal(window.CPOIDEHooks.parsePyret, "surface-parse");
+    // TODO(joe): is "this" trustworthy here?
+    var self = this;
+    var parse = this._getVal("parsePyret", "surface-parse");
```

Comment:
I think you're better off making `_getVal` a module-level function that doesn't get exported. Then you don't need to worry about `this` and you don't need to worry about anyone monkeying around with `_getVal`.